src/serac/physics/utilities/functional/domain_integral_cuda.cuh

```diff
@@ -0,0 +1,259 @@
+#pragma once
+
+#include "mfem.hpp"
+#include "mfem/linalg/dtensor.hpp"
+
+#include "serac/physics/utilities/functional/integral_utilities.hpp"
+#include "serac/physics/utilities/functional/domain_integral_shared.hpp"
+#include <cstring>
+
+namespace serac {
+
+namespace detail {
+
+/**
+ * @brief utility method to display last cuda error message
+ *
+ * @param[in] o The output stream to post success or CUDA error messages
+ * @param[in] success_string A string to print if there are no CUDA error messages
+ */
+inline void displayLastCUDAErrorMessage(std::ostream& o, const char* success_string = "")
```

Comment:
Another generic function that is deep in the weeds of functional

Reply:
probably in accelerator.hpp / accelerator.cpp?